### PR TITLE
Publish prebuilt reports for each repo

### LINKED ISSUE #4160: Prebuilt reports for each repo should be stored as pipeline artifacts in CI builds
When investigating prebuilts, it can be necessary to inspect each individual repo's prebuilt report rather than the overall build's prebuilt report. This is especially necessary when the prebuilts are not associated with a given project (an example of this is the SDK loading of a targeting pack). In that case, the easiest way to identify which repo is causing that prebuilt is to look at each repo's prebuilt report.

The problem is that the CI build of the VMR doesn't include this repo prebuilt reports in the pipeline artifacts. So its necessary to run a local build with the same commit in order to have access to that information. It would be better if the CI build would store these as pipeline artifacts. This would make investigations go more quickly.

### PR DESCRIPTION
**Pull request by Winniexu01**
Fixes https://github.com/dotnet/source-build/issues/4160

Store the prebuilt repots for each repo as pipeline artifacts rather than the overall build's prebuilt report, which will make investigations of prebuilt go more quickly.

I introduced a new property `DotNetBuildPrebuiltReportDir` which point to `artifacts\prebuilt-report\<reponame>\`, all repo's prebuilt reports will respect this property and move to that folder.

### CODE REVIEW

**Comment by mthalman** [comment]
@Winniexu01 - Is there an example build containing a prebuilt that you can provide which demonstrates this new behavior?

**Comment by Winniexu01** [comment]
@mthalman Example build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2536531&view=results (internal Microsoft link)

**Comment by Winniexu01** [comment]
Hi @mthalman Here is [new build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2537646) (internal Microsoft link) that CentOS9 failed with '3 Prebuilt Exist'.

**Review by mthalman — APPROVED** [review]

File: src/SourceBuild/content/repo-projects/Directory.Build.targets [hunk 1]
```diff
@@ -511,18 +511,24 @@
...
   </Target>
 
-  <Target Name="MoveDotNetBuildArtifactsLogs"
+  <Target Name="MoveDotNetBuildLogs"
         Condition="'$(IsUtilityProject)' != 'true' and
                     Exists('$(ProjectDirectory)artifacts')">
     <ItemGroup>
       <LogFilesToMove Include="$(ProjectDirectory)artifacts/**/*.log" />
       <LogFilesToMove Include="$(ProjectDirectory)artifacts/**/*.binlog" />
+      <PrebuiltReportsToMove Include="$(ProjectDirectory)artifacts/**/*usage.xml" />
```
[the remark this answers is not in the document]
Comment (Winniexu01) [reply src/SourceBuild/content/repo-projects/Directory.Build.targets]: Sure. I updated the related code.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**